src/software/new_geom/util/BUILD

```diff
@@ -1,198 +1,112 @@
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
-    name = "util",
-    deps = [
-        ":closest_point",
-        ":distance",
-        ":intersects",
-    ],
-)
-
-cc_library(
-    name = "closest_point",
+    name = "algorithms",
     srcs = [
+        "almost_equal.cpp",
         "closest_point.cpp",
+        "collinear.cpp",
+        "contains.cpp",
+        "distance.cpp",
+        "find_open_circles.cpp",
+        "furthest_point.cpp",
+        "intersection.cpp",
+        "intersects.cpp",
+        "voronoi_diagram.cpp",
```

Comment:
so I've done some work that shows that only voronoi needs boost geom and the rest can do without. This is gonna unnecessarily increase build times if group voronoi in algorithms

Reply:
I think that higher level abstract algorithms should be in their own targets

Reply:
I don't think this should increase build times? Boost slows things down only if you include a boost header somewhere that gets includes transitively in lots of other headers. The build time for this should be effectively exactly the same.

Reply:
there are so places in the code base that only need distance or contains or intersects, but not findopencircles, so building those targets will take way longer

Reply:
I suppose those targets, if built _entirely independently_ might take slightly longer to build, but in the context of things like full system or full tests, there will be no performance hit, since we're already building these files. For those particular cases that would take longer, it really would only be a few seconds, if it was a totally clean build with nothing cached. 

(_definitely don't want any significant performance regressions, I just don't think this is one_)